kura/org.eclipse.kura.rest.inventory.provider/src/main/java/org/eclipse/kura/internal/rest/inventory/InventoryRestService.java

```diff
@@ -0,0 +1,311 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.internal.rest.inventory;
+
+import static org.eclipse.kura.cloudconnection.request.RequestHandlerMessageConstants.ARGS_KEY;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.cloudconnection.message.KuraMessage;
+import org.eclipse.kura.cloudconnection.request.RequestHandler;
+import org.eclipse.kura.cloudconnection.request.RequestHandlerRegistry;
+import org.eclipse.kura.core.inventory.InventoryHandlerV1;
+import org.eclipse.kura.message.KuraPayload;
+import org.eclipse.kura.request.handler.jaxrs.DefaultExceptionHandler;
+import org.eclipse.kura.request.handler.jaxrs.JaxRsRequestHandlerProxy;
+import org.osgi.service.useradmin.Role;
+import org.osgi.service.useradmin.UserAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/inventory/v1")
+public class InventoryRestService {
+
+    private static final Logger logger = LoggerFactory.getLogger(InventoryRestService.class);
+
+    private static final String APP_ID = "INVEN-V1";
+
+    private static final String KURA_PERMISSION_REST_CONFIGURATION_ROLE = "kura.permission.rest.inventory";
+
+    private final RequestHandler requestHandler = new JaxRsRequestHandlerProxy(this);
```

Comment:
The request handler for `INVENTORY-V1` is already implemented by the original `InventoryHandlerV1`. This can be removed together with the `APP_ID` constant, the `setRequestHandlerRegistry` and `unsetRequestHandlerRegistry` methods and the `RequestHandlerRegistry` reference in DS component definition.